src/Verbs/PushCommand.cs

```diff
@@ -0,0 +1,179 @@
+            { 
+                packageHelper.EnsurePackageDoesntAlreadyExists(packages, filePath);
+            }
+
+            // Parse package.xml before upload to print out info
+            // and to use for comparisson on what is already uploaded
+            var packageInfo = Parse.PackageXml(filePath);
+
+            // OK all checks passed - time to upload it
+            await UploadPackage(options);
+
+            // Got this far then it got uploaded to our.umb all OK
+            Console.WriteLine(Resources.Push_Complete, filePath);
+
+            return 0;
+        }
+
+        private static async Task UploadPackage(PushOptions options)
+        {
+            try
+            {
+                // HttpClient will use this event handler to give us
+                // Reporting on how its progress the file upload
+                var processMsgHander = new ProgressMessageHandler(new HttpClientHandler());
+                processMsgHander.HttpSendProgress += (sender, e) =>
+                {
+                    // Could try to reimplement progressbar - but that library did not work in GH Actions :(
+                    var percent = e.ProgressPercentage;
+                };
+
+                var packageHelper = new PackageHelper();
+
+                Console.Write(Resources.Push_Uploading, Path.GetFileName(options.Package));
+
+                using (var client = packageHelper.GetClientBase(options.ApiKey))
+                {
+                    MultipartFormDataContent form = new MultipartFormDataContent();
+                    var fileInfo = new FileInfo(options.Package);
+                    var content = new StreamContent(fileInfo.OpenRead());
+                    content.Headers.ContentDisposition = new ContentDispositionHeaderValue("form-data")
+                    {
+                        Name = "file",
+                        FileName = fileInfo.Name
+                    };
+                    form.Add(content);
+                    form.Add(new StringContent(ParsePublishFlag(options.Publish)), "isCurrent");
+                    form.Add(new StringContent(options.DotNetVersion), "dotNetVersion");
+                    form.Add(new StringContent("package"), "fileType");
+                    form.Add(GetVersionCompatability(options.WorksWith), "umbracoVersions");
+
+                    var httpResponse = await client.PostAsync("/Umbraco/Api/ProjectUpload/UpdatePackage", form);
+                    if (httpResponse.StatusCode == HttpStatusCode.Unauthorized)
+                    {
+                        packageHelper.WriteError(Resources.Push_ApiKeyInvalid);
+                        Environment.Exit(5); // ERROR_ACCESS_DENIED
+                    }
+                    else if (httpResponse.IsSuccessStatusCode)
+                    {
+                        Console.WriteLine(Resources.Push_Complete);
+                        var apiReponse = await httpResponse.Content.ReadAsStringAsync();
```

Comment:
Are we going to use this? Commented out on the next line 🙂

Reply:
commented out for now.